src/memgraph.cpp

```diff
@@ -676,6 +677,19 @@ void InitSignalHandlers(const std::function<void()> &shutdown_fun) {
             "Unable to register SIGINT handler!");
 }
 
+std::string GetMachineId() {
+#ifdef MG_TELEMETRY_ID_OVERRIDE
+  return MG_TELEMETRY_ID_OVERRIDE;
+#else
+  // We assume we're on linux and we need to read the machine id from /etc/machine-id
+  const auto machine_id_lines = memgraph::utils::ReadLines("/etc/machine-id");
+  if (machine_id_lines.size() != 1) {
+    return "UNKNOWN";
+  }
+  return machine_id_lines[0];
+#endif
+}
```

Comment:
I was thinking about moving this under utils but it has the telemetry-related `ifdef` + it's composed of simple units of work (`Readlines`) so it's fine, but do we add an e2e test that verifies `machine_id` is present (if it's not already there)?

Reply:
I am adding e2e right now, to have verification that everything works.

Reply:
I have moved it to the utils, Not sure if it makes sense to add tests for this system info stuff, what do you think, not sure how to confirm correct results or mock this stuff.

Reply:
Yea it's platform-specific, I was thinking about that, maybe we can add a simple unit test that can be run on each platform we are building for (because even on different Linux machines this can be different + once we'll add support for Win/Mac, we'll also have to make this code multi-platform.

I think for now it's not worth investing much time into testing / cross-platform support, we'll get there.